indico/modules/oauth/controllers.py

```diff
@@ -16,14 +16,130 @@
+    @oauth.authorize_handler
+    def _process(self, **kwargs):
+        if request.method == 'POST':
+            return 'confirm' in request.form
+        if self.application.is_trusted:
+            return True
+        return render_template('oauth/authorize.html', application=self.application)
+
+
+class RHOAuthToken(RH):
+    @oauth.token_handler
+    def _process(self, **kwargs):
+        return None
+
+
+class RHOAuthAdmin(RHAdminBase):
+    """OAuth server administration settings"""
+
+    def _process(self):
+        applications = OAuthApplication.find().order_by(db.func.lower(OAuthApplication.name)).all()
+        return WPOAuthAdmin.render_template('apps.html', applications=applications)
+
+
+class RHOAuthAdminApplicationBase(RHAdminBase):
+    """Base class for single OAuth application RHs"""
+    def _checkParams(self):
+        self.application = OAuthApplication.get(request.view_args['id'])
+
+
+class RHOAuthAdminApplication(RHOAuthAdminApplicationBase):
+    """Handles application details page"""
+
+    def _process(self):
+        form = ApplicationForm(obj=self.application, application=self.application)
+        if form.validate_on_submit():
+            form.populate_obj(self.application)
+            flash(_("Application {} was modified").format(self.application.name), 'success')
+            return redirect(url_for('.apps'))
+        return WPOAuthAdmin.render_template('app_details.html', application=self.application, form=form)
+
+
+class RHOAuthAdminApplicationDelete(RHOAuthAdminApplicationBase):
+    """Handles OAuth application deletion"""
+
+    def _process(self):
+        db.session.delete(self.application)
+        logger.info("Application {} was deleted.".format(self.application))
+        flash(_("Application deleted successfully"), 'success')
+        return redirect(url_for('.apps'))
+
+
+class RHOAuthAdminApplicationNew(RHAdminBase):
+    """Handles OAuth application registration"""
+
+    def _process(self):
+        form = ApplicationForm()
+        if form.validate_on_submit():
+            application = OAuthApplication()
+            form.populate_obj(application)
+            db.session.add(application)
+            flash(_("Application {} registered successfully").format(application.name), 'success')
+            return redirect(url_for('.apps'))
+        return WPOAuthAdmin.render_template('app_new.html', form=form)
+
+
+class RHOAuthAdminApplicationReset(RHOAuthAdminApplicationBase):
+    """Resets the client secret of the OAuth application"""
+
+    def _process(self):
+        self.application.reset_client_secret()
+        flash(_("New client secret generated for the application"), 'success')
+        return redirect(url_for('.app_details', self.application))
+
+
+class RHOAuthAdminApplicationRevoke(RHOAuthAdminApplicationBase):
+    """Revokes all user tokens associated to the OAuth application"""
+
+    def _process(self):
+        OAuthToken.find(application_id=request.view_args['id']).delete()
```

Comment:
`self.application.tokens.delete()` assuming it's lazy=dynamic
i'd also log the `self.application` instead of just the id